build/template_kql/common/extend_resource.kql

```diff
@@ -398,10 +398,25 @@ resources
     properties
 | join kind = leftouter (
     recoveryservicesresources
-    | where name startswith "VM;"
-    | project Name = tolower(tostring(properties.dataSourceInfo.resourceID)), Backup = tostring(properties.protectionStatus)
+    | extend vmId = case(
+        name startswith "VM;", tolower(tostring(properties.dataSourceInfo.resourceID)),
```

Comment:
Check possibility of replacing below condition
name startswith "VM;" to  properties.backupManagementType == "AzureIaasVM" in app places where we want to pick VM protection information

Reply:
According to the API reference, this value will be fixed, so we can use it.
https://learn.microsoft.com/en-us/azure/templates/microsoft.recoveryservices/vaults/backuppolicies?pivots=deployment-language-bicep#protectionpolicy

Reply:
Fixed.